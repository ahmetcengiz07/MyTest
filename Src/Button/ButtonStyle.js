import {StyleSheet} from 'react-native';

export const styles = {
  primary: StyleSheet.create({
    container: {
      backgroundColor: 'lightblue',
      justifyContent: 'center',
      alignItems: 'center',
      margin: 20,
      padding: 20,
      borderRadius: 50,
    },
    text: {
      fontSize: 30,
      fontWeight: 'bold',
    },
  }),
  secondary: StyleSheet.create({
    container: {
      borderColor: 'lightblue',
      borderWidth: 1,
      justifyContent: 'center',
      alignItems: 'center',
      margin: 20,
      padding: 20,
      borderRadius: 50,
    },
    text: {
      fontSize: 30,
      fontWeight: 'bold',
    },
  }),
};
