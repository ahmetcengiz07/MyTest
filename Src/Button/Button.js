import {Text, TouchableOpacity} from 'react-native';
import React from 'react';
import {styles} from './ButtonStyle';

const Button = ({title, onPress, theme = 'primary'}) => {
  //themenin default deÄŸerini primary olarak belirledik
  return (
    <TouchableOpacity
      testID="button-touchable"
      onPress={onPress}
      style={styles[theme].container}>
      <Text testID="button-title" style={styles[theme].title}>
        Testing Style
      </Text>
    </TouchableOpacity>
  );
};

export default Button;
