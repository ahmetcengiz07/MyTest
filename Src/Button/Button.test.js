import React from 'react';
import {render, fireEvent, cleanup} from '@testing-library/react-native';
import Button from './Button';
import {styles} from './ButtonStyle';

beforeAll(() => {
  console.log('Bir kere çalış arkasından diğer testler çalışssın ');
});

beforeEach(() => {
  console.log('Her testten önce bir kez çalış');
});

afterEach(() => {
  console.log('Her testten sonra bir kez çalış');
});

afterAll(() => {
  console.log('Tüm testlerden sonra bir kez çalış');
});
//cleanup() fonksiyonu kullanama gerek yok çünkü afterAll içerisinde mevcut

test('should match with snapshot', () => {
  const comp = render(<Button />);
  expect(comp).toMatchSnapshot(); 
  //Cevap almak istenilen nesneyi parametre olarak alır.Burada toMatchSnapshot koşulunu sağlayan nesne comp mudur ?
});

test('Check Number', () => {
  expect(5 + 2).toBe(7);
});
//Button komponenti ile oluşturuluna texte gönderilen title parametresi Button komponenti içersindeki text de yazan string ifade ile aynı mı değil mi ?
//koşul kontrolünü jest yapar , Komponentin renderını da testing-library yapar.
test('should render title correctly', () => {
  const testTitle = 'test';
  const comp = render(<Button title={testTitle} />);
  const buttonText = comp.getByTestId('button-title').children[0];
  console.log(buttonText);
});

test('should be trigged onPress', () => {
  const mockFunction = jest.fn(); //fonks ismi önemli değil jest.fn() ile mockFunctionun her çağırıldığında çalışıp çalışmadığını görebiliriz
  const comp = render(<Button onPress={mockFunction} />);
  const buttonTouchable = comp.getByTestId('button-touchable');
  fireEvent(buttonTouchable, 'Press'); //user eventi (tıklama) kontrolü
  expect(mockFunction).toBeCalledTimes(1); //kontrolünü yaptığımız yer
});

test('should render given or default theme style', () => {
  const selectedTheme = 'primary';
  const comp = render(<Button theme={selectedTheme} />);
  const buttonContainer = comp.getByTestId('button-touchable').props.style;
  expect(buttonContainer).toMatchObject(styles[selectedTheme].container);
  //verilen theme başarılı bir şekilde buttonun içerisine verilmiş
});
