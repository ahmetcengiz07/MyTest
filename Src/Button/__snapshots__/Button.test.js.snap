// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "lightblue",
      "borderRadius": 50,
      "justifyContent": "center",
      "margin": 20,
      "opacity": 1,
      "padding": 20,
    }
  }
  testID="button-touchable"
>
  <Text
    testID="button-title"
  >
    Testing Style
  </Text>
</View>
`;
